File: Management/Models/CanvasModels/Assignments/CanvasRubricAssociation.cs
namespace CanvasModel.Assignments;

public record CanvasRubricAssociation
{
  [JsonPropertyName("id")]
  public ulong Id { get; set; }

  [JsonPropertyName("rubrid_id")]
  public ulong RubricId { get; set; }

  [JsonPropertyName("association_id")]
  public ulong AssociationId { get; set; }

  [JsonPropertyName("association_type")]
  public required string AssociationType { get; set; }

  [JsonPropertyName("use_for_grading")]
  public bool UseForGrading { get; set; }

  [JsonPropertyName("summary_data")]
  public string? SummaryDaata { get; set; }

  [JsonPropertyName("purpose")]
  public required string Purpose { get; set; }

  [JsonPropertyName("hide_score_total")]
  public bool? HideScoreTotal { get; set; }

  [JsonPropertyName("hide_points")]
  public bool HidePoints { get; set; }

  [JsonPropertyName("hide_outcome-results")]
  public bool HideOUtcomeResult { get; set; }

}

File: src/features/canvas/models/assignments/canvasRubricAssociation.ts
export interface CanvasRubricAssociation {
  id: number;
  rubric_id: number;
  association_id: number;
  association_type: string;
  use_for_grading: boolean;
  summary_data?: string;
  purpose: string;
  hide_score_total?: boolean;
  hide_points: boolean;
  hide_outcome_results: boolean;
}
